### PR TITLE
Address Summoner Elemental Casting Time - and Light Spirit Idle

### PR DESCRIPTION
**Pull request by hargrave81**
This supports falling back to the existing "monster 24s casting mode" if too many players do not like the retail nature.

The petcontroller was modified to handle elementals and support controlling to some extent the actions speed/and or spells via lua script. Some modification to scope for the mob controller was necessary, though further clean up could change that. This was just V1.

#3561 SMN Spirit flaws issue

What changed? Elementals were casting based on the monster's default cooldown of about 24 seconds. They also "opened up" casting almost all the time on a "sic" command. Now elementals will cast based on SMN skill (base of 45s at skill cap, faster with higher than cap, slower with lower than cap). Weather and other modifiers were added as well.

Light Spirit now casts buffs on master and party members, as well as heals them as needed.

AI for light spirit is heavily contained within lua script onRoam. The default casting AI is used during battle for light spirit, it will randomly cure v.s. attack.

TODO: I have not had much time testing, so lots of testing, make sure **other pets** are not negatively impacted. Also put in additional code in lua script to read the gear smn skill bonus gear. I don't think the default helper does that. Fix the counter from being reset everytime the monster re-engages the monster.

Other Notes: 
- If there's going to be unique AI for each spirit, then the AI needs to be broken out into individual files.  Potentially this would be a nice upgrade to allow each spirit their own AI (healing v.s. non-healing or custom spirits)
- Clean up lua scripts to be more readable and unify the elemental (dsp.elements.WIND for example) and the database having a different number than other element Enums.

Reference https://github.com/DarkstarProject/darkstar/pull/5966 for other additional notes.

### CODE REVIEW

**Comment by hargrave81** [comment]
@ibm2431 , @teschnei - I have cleaned up the code commits.  I also cleaned up the boolean default to false noted.  I updated the notes for the pull request as well.

~Chris

File: scripts/globals/pets/avatar.lua [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        return sms * - 1
+    end
+    return 0
+end;
+
+function getDayMod(avatar)
+    local element = getElement(avatar)
+    local ms = -3000    
+    local summoner = avatar:getMaster()    
+    if VanadielDayElement() == dsp.day.ICEDAY and element == dsp.subEffect.ICE_DAMAGE then
```
Comment (TeoTwawki) [inline scripts/globals/pets/avatar.lua]: tables my dude. tables.
```lua
local myTableNameHere =
{
    [index here] = what value we want from selecting it,
    [index here] = what value we want from selecting it,
    [index here] = what value we want from selecting it
}

if myTable[input to get index here] then
```

Comment (TeoTwawki) [reply scripts/globals/pets/avatar.lua]: pretty much anytime you have to type "else" that many times, you are working to hard

Comment (TeoTwawki) [reply scripts/globals/pets/avatar.lua]: nvm, didn't realize this was your other PR re sent in, I expected the new commits to replace what was on the other one (github will auto update the pr if the branch's content changes, so force pushing to overwrite it works)

**Comment by teschnei** [comment]
agh, all my comments are gone

it's better to force push over an existing branch with a PR open cause it'll keep the PR open but replace the commits with the new ones

**Comment by teschnei** [comment]
@ibm2431 you wanna improve the scripts in here and let me do the core?

**Comment by ibm2431** [comment]
@teschnei Yeah, I can help out!

(May not be able to start working on it today though; it's getting late for me.)

**Comment by ibm2431** [comment]
https://github.com/hargrave81/darkstar/pull/5

**Comment by helixhamin** [comment]
@hargrave81 
Please don't change default settings like that. The ver lock is set to 1 because version mismatch can cause problems. The server owner can change it themselves if they want. If they don't understand enough to change it, then what do you think will happen when version mismatch causes strange behavior.

File: version.info [hunk 2]
```diff
@@ -6,7 +6,7 @@ CLIENT_VER: 30190328_2
 #0 - disabled (every version allowed)
 #1 - enabled - strict (only exact CLIENT_VER allowed, default)
 #2 - enabled - greater than or equal  (matching or greater than CLIENT_VER allowed)
-VER_LOCK: 1
+VER_LOCK: 2
```
Comment (helixhamin) [inline version.info]: Should not be changed...

**Comment by hargrave81** [comment]
I'm not at my pc, I'll revert this.  Forgot this would change the pull request on the fly.  Had a client that couldn't connect because POL update is newer.  I'll change my deploy scripts to modify that during deploy.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by teschnei** [comment]
I pushed the changes (minus the version.info commit) manually to the smn-spirits branch, so this is on our repo now. @ibm2431 can you PR to that branch instead? also, I fixed getFamily, but I have to wait for someone to merge it before I can put it on the branch

**Pull request closed without merging**